### DISCUSSION TITLE
How can I improve my attribute sorting?

### DISCUSSION THREAD

**Discussion by kaangokdemir** [prettier/plugin-pug#190] [Q&A]
## Question

Hey everyone. I have a  pug element like below and want to order it like the "Expected Output". My dream ordering is:

1- `v-if`, `v-else` and `v-else-if`
2- `v-model`
3-  Any attribute starts with `v-`
4- `v-for` (It should always come after the other `v-`'s)
5- `:key` to see `v-for` and `:key` always together.
6- Any attribute starts with `@`
7- Any attribute starts with `:`
8- Attributes starts with `ref`
9- Static attributes which has a value like `type='text'`
10- Any other static attributes which doesn't have a value like `disabled`, `required` etc.

Plus: On same priority level, always sort alphabetically.

I wonder currently this plugin satisfies that or need to develop a feature for it. Thank you!

## Input

```pug
div(
  v-else='blabla',
  v-else-if='blalba',
  v-if='blabla',
  v-kaan='kaan',
  v-model='blabla',
  v-zet='blabla',
  v-for='item in list',
  @chris='chris',
  @click='click',
  @kaan='kaan',
  @submit='submit',
  :chris='chris',
  :kaan='kaan',
  :key='key',
  :ref='ref',
  :zet='zet',
  ref='ref',
  alpha='kaan',
  autocomplete,
  bravo='braavo',
  chris,
  kaan,
  zet='zet')
```

## Expected Output

```pug
div(
  v-else='blabla',
  v-else-if='blalba',
  v-if='blabla',
  v-kaan='kaan',
  v-model='blabla',
  v-zet='blabla',
  v-for='item in list',
  :key='key',
  @chris='chris',
  @click='click',
  @kaan='kaan',
  @submit='submit',
  :chris='chris',
  :kaan='kaan',
  :ref='ref',
  :zet='zet',
  ref='ref',
  alpha='kaan',
  bravo='braavo',
  zet='zet'
  autocomplete,
  chris,
  kaan)
```

## Additional Context

My current `.prettierrc` file (Without comments) like below. Even it helps a lot to reach my target, it still not fully works and some of my expectations not satisfies like putting `:key` behind `v-for` and putting static attributes with a value above the attributes without value like `type="text"` over `disabled`

```jsonc
{
  "semi": false,
  "arrowParens": "always",
  "singleQuote": true,
  "closingBracketPosition": "last-line",
  "pugSortAttributesBeginning": [
    "^v-else$",
    "^v-if$",
    "^v-else-if$",
    "^v-model",
    "^v-",
    "^v-for$", //v-for should be the last attribute which starts with v-
    "^:key$", // key binding should always follow v-for
    "^@",
    "^:", //any shortcut for v-bind
    "^ref$",
    "='", //attributes which has a value in it like foo='bar'
  ],
  "pugSortAttributes": "asc"
}
```

**Comment by Shinigami92** [comment]
@SkyaTura do you want to have a look into this? If I remember correctly, you provided this feature :slightly_smiling_face:

**Comment by SkyaTura** [comment]
Currently is not possible to differentiate between attributes that have or have not specified values (or better saying, between strings and booleans) because during sort this information isn't available.

Also, I believe there is a bug that prevents `:key` to be in the correct place. I'll further investigate this issue.

**Comment by kaangokdemir** [comment]
Do we have any progress on that? @Shinigami92 @SkyaTura

**Reply by Shinigami92** [reply]
Currently no, I'm on some other tasks that are necessary and get bundled in the upcoming 1.14
If you want and speed up the process, feel free to open and start on a PR 🙂
